tools/nvqpp/nvq++.in

```diff
@@ -584,6 +607,16 @@ trap delete_temp_files EXIT
 
 COMPILER_FLAGS="${CPPSTD} -Wno-c++20-extensions ${COMPILER_FLAGS}"
 
+# If the user has requested NVQC, choose a reasonable default. Note that Nvidia
+# GPUs do not need to be installed in order to select an Nvidia-powered backend
+# when using NVQC.
+if ${NVQC}; then
+	NVQIR_SIMULATION_BACKEND="custatevec-fp32"
```

Comment:
I don't see that this would be overridden before it was used, and hence this doesn't look like  a default so much as that this will always be used. What am I missing?

Reply:
> I don't see that this would be overridden before it was used, and hence this doesn't look like a default so much as that this will always be used. What am I missing?

If the user specifies a target with `--target <the_target>`, then `TARGET_CONFIG` will be set, and line 628 below would source the corresponding target config file. This would override `NVQIR_SIMULATION_BACKEND`.

The main reason for this new piece of code is to hosts without GPUs to invoke NVQC w/o accidentally telling NVQC to use the `qpp-cpu` backend (which is the default on a machine without GPUs).